File: VB/Creational/Prototype/ConcretePrototype.vb
Imports System.Net.Mime

Class ConcretePrototype
	Implements IPrototype

	Public Name As String
	Public Number As Integer
	Private Guid As String

	Public Sub New(name As String, number As Integer)
		Me.Name = name
		Me.Number = number
		Me.Guid = System.Guid.NewGuid().ToString()
	End Sub

	Public Overrides Function ToString() As String
		Return ("I am Prototype " + Guid + " - " + Name + " - # " + Number.ToString())
	End Function

	Public Function Clone() As IPrototype Implements IPrototype.Clone
		Return Me.MemberwiseClone()
	End Function
End Class

File: Creational/Prototype/ConcretePrototype.cs
// See https://aka.ms/new-console-template for more information
public class ConcretePrototype : IPrototype
{
	public string Name { get; set; }
	public int Number	{ get; set; }
	private Guid guid { get; set; }

	public ConcretePrototype(string name, int number)
	{
		Name = name;
		Number = number;
		guid = Guid.NewGuid();
	}
	public ConcretePrototype(ConcretePrototype firstPrototype)
	{
		this.Name = firstPrototype.Name;
		this.Number = firstPrototype.Number;
		this.guid = firstPrototype.guid;
	}

	public string ToString()
	{
		return "public Name: " + Name + " - public Number: " + Number + " - private guid: " + guid;
	}

	public Object Clone()
	{
		return (IPrototype)this.MemberwiseClone();
	}
}
